core/drivers/gic.c

```diff
@@ -248,8 +264,11 @@ static void gic_it_enable(struct gic_data *gd, size_t it)
 
 	/* Assigned to group0 */
 	assert(!(read32(gd->gicd_base + GICD_IGROUPR(idx)) & mask));
-	/* Not enabled yet */
-	assert(!(read32(gd->gicd_base + GICD_ISENABLER(idx)) & mask));
+	if (it >= NUM_SGI)
+	/* Not enabled yet, except Software Generated Interrupt
```

Comment:
rather not have comments between if `if (...)` and the related execution block.
either braces or add comment before the `if()`.